cadquery/occ_impl/shapes.py

```diff
@@ -1034,6 +1036,85 @@ def intersect(self, *toIntersect: "Shape") -> "Shape":
 
         return self._bool_op((self,), toIntersect, intersect_op)
 
+    def facesIntersectedByLine(
+        self,
+        point: VectorLike,
+        axis: VectorLike,
+        tol: float = 1e-4,
+        direction: Optional[Literal["AlongAxis", "Opposite"]] = None,
+    ):
+        """
+        Computes the intersections between the provided line and the faces of the provided shape
+
+        :point: Base point for defining a line
+        :axis: Axis on which the line rest
+        :tol: Intersection tolerance
+        :direction: Valid values : "AlongAxis", "Opposite", if specified will ignore all faces that are not in the specified direction
+        including the face where the :point: lies if it is the case
+
+        :returns: A list of intersected faces sorted by distance from :point:
+        """
+
+        oc_point = (
+            gp_Pnt(*point.toTuple()) if isinstance(point, Vector) else gp_Pnt(*point)
+        )
+        oc_axis = (
+            gp_Dir(Vector(axis).wrapped)
+            if not isinstance(axis, Vector)
+            else gp_Dir(axis.wrapped)
+        )
+
+        line = gce_MakeLin(oc_point, oc_axis).Value()
+        shape = self.wrapped
+
+        intersectMaker = BRepIntCurveSurface_Inter()
+        intersectMaker.Init(shape, line, tol)
+
+        faces_dist = []  # using a list instead of a dictionnary to be able to sort it
+        while intersectMaker.More():
+            interPt = intersectMaker.Pnt()
+            interDirMk = gce_MakeDir(oc_point, interPt)
+
+            distance = oc_point.SquareDistance(interPt)
+
+            # interDir is not done when `oc_point` and `oc_axis` have the same coord
+            if interDirMk.IsDone():
+                interDir: Any = interDirMk.Value()
+            else:
+                interDir = None
+
+            if direction == "AlongAxis":
+                if (
+                    interDir is not None
+                    and not interDir.IsOpposite(oc_axis, tol)
+                    and distance > tol
+                ):
+                    faces_dist.append((intersectMaker.Face(), distance))
+
+            elif direction == "Opposite":
+                if (
+                    interDir is not None
+                    and interDir.IsOpposite(oc_axis, tol)
+                    and distance > tol
+                ):
+                    faces_dist.append((intersectMaker.Face(), distance))
+
+            elif direction is None:
+                faces_dist.append(
+                    (intersectMaker.Face(), abs(distance))
+                )  # will sort all intersected faces by distance whatever the direction is
```

Comment:
I think you can remove this `direction=None` case, I can't see anywhere you use it.

Reply:
The purpose of this is to be more general. It's not used in the PR but if someone wants to get a list of all the faces intersected by a LIne without consideration of the direction then it will be used.
That's why I didn't made this method private. I'm not sure it will be used that way one day though. I'm find with removing it and making the method private, you tell me

Reply:
I don't mind if the method is private or not, but `direction=None` doesn't have any tests. So I would say remove it or test it please.